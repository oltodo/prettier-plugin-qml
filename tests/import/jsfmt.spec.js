run_spec(__dirname, ["qml"]);
