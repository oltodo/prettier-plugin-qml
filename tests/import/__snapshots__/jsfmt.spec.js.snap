// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import.json - qml-verify: import.json 1`] = `
{
  "kind": "Program",
  "loc": {
    "start": {
      "offset": 0,
      "line": 1,
      "column": 1
    },
    "end": {
      "offset": 758,
      "line": 31,
      "column": 1
    }
  },
  "children": [
    {
      "kind": "Import",
      "loc": {
        "start": {
          "offset": 0,
          "line": 1,
          "column": 1
        },
        "end": {
          "offset": 20,
          "line": 1,
          "column": 18
        }
      },
      "identifier": "QtQuick",
      "version": "2.5"
    },
    {
      "kind": "Import",
      "loc": {
        "start": {
          "offset": 21,
          "line": 2,
          "column": 1
        },
        "end": {
          "offset": 60,
          "line": 2,
          "column": 34
        }
      },
      "identifier": "QtQuick.Window",
      "version": "2.2",
      "as": "Window"
    },
    {
      "kind": "Import",
      "loc": {
        "start": {
          "offset": 64,
          "line": 3,
          "column": 1
        },
        "end": {
          "offset": 106,
          "line": 3,
          "column": 36
        }
      },
      "path": "path/to/library.js  ",
      "as": "Library"
    },
    {
      "kind": "Import",
      "loc": {
        "start": {
          "offset": 107,
          "line": 4,
          "column": 1
        },
        "end": {
          "offset": 137,
          "line": 4,
          "column": 8
        }
      },
      "path": "   path/to/components"
    },
    {
      "kind": "ObjectDefinition",
      "identifier": "Rectangle",
      "children": [
        {
          "kind": "Property",
          "loc": {
            "start": {
              "offset": 155,
              "line": 7,
              "column": 5
            },
            "end": {
              "offset": 178,
              "line": 7,
              "column": 18
            }
          },
          "type": "var",
          "identifier": "helloWorld"
        },
        {
          "kind": "Property",
          "loc": {
            "start": {
              "offset": 187,
              "line": 9,
              "column": 8
            },
            "end": {
              "offset": 211,
              "line": 9,
              "column": 31
            }
          },
          "type": "var",
          "identifier": "someNumber",
          "value": {
            "kind": "JavascriptBlock",
            "loc": {
              "start": {
                "offset": 212,
                "line": 9,
                "column": 33
              },
              "end": {
                "offset": 215,
                "line": 9,
                "column": 33
              }
            },
            "value": "1.5"
          }
        },
        {
          "kind": "Property",
          "loc": {
            "start": {
              "offset": 220,
              "line": 10,
              "column": 5
            },
            "end": {
              "offset": 243,
              "line": 10,
              "column": 27
            }
          },
          "type": "var",
          "identifier": "someList",
          "value": {
            "kind": "JavascriptBlock",
            "loc": {
              "start": {
                "offset": 244,
                "line": 10,
                "column": 29
              },
              "end": {
                "offset": 267,
                "line": 10,
                "column": 51
              }
            },
            "value": "[1, 2, \\"three\\", \\"four\\"]"
          }
        },
        {
          "kind": "Property",
          "loc": {
            "start": {
              "offset": 272,
              "line": 11,
              "column": 5
            },
            "end": {
              "offset": 296,
              "line": 11,
              "column": 28
            }
          },
          "type": "var",
          "identifier": "someObject",
          "value": {
            "kind": "ObjectDefinition",
            "identifier": "Rectangle",
            "children": [
              {
                "kind": "Attribute",
                "identifier": "width",
                "value": {
                  "kind": "JavascriptBlock",
                  "loc": {
                    "start": {
                      "offset": 316,
                      "line": 11,
                      "column": 49
                    },
                    "end": {
                      "offset": 320,
                      "line": 11,
                      "column": 52
                    }
                  },
                  "value": "100;"
                }
              },
              {
                "kind": "Attribute",
                "identifier": "height",
                "value": {
                  "kind": "JavascriptBlock",
                  "loc": {
                    "start": {
                      "offset": 329,
                      "line": 11,
                      "column": 62
                    },
                    "end": {
                      "offset": 333,
                      "line": 11,
                      "column": 65
                    }
                  },
                  "value": "100;"
                }
              },
              {
                "kind": "Attribute",
                "identifier": "color",
                "value": {
                  "kind": "JavascriptBlock",
                  "loc": {
                    "start": {
                      "offset": 341,
                      "line": 11,
                      "column": 74
                    },
                    "end": {
                      "offset": 346,
                      "line": 11,
                      "column": 74
                    }
                  },
                  "value": "\\"red\\""
                }
              }
            ]
          }
        },
        {
          "kind": "Property",
          "loc": {
            "start": {
              "offset": 353,
              "line": 12,
              "column": 5
            },
            "end": {
              "offset": 380,
              "line": 12,
              "column": 31
            }
          },
          "type": "color",
          "identifier": "someColor",
          "value": {
            "kind": "JavascriptBlock",
            "loc": {
              "start": {
                "offset": 383,
                "line": 12,
                "column": 35
              },
              "end": {
                "offset": 389,
                "line": 12,
                "column": 35
              }
            },
            "value": "\\"blue\\""
          }
        },
        {
          "kind": "Property",
          "loc": {
            "start": {
              "offset": 419,
              "line": 15,
              "column": 5
            },
            "end": {
              "offset": 451,
              "line": 15,
              "column": 36
            }
          },
          "default": true,
          "type": "var",
          "identifier": "someString",
          "value": {
            "kind": "JavascriptBlock",
            "loc": {
              "start": {
                "offset": 452,
                "line": 15,
                "column": 38
              },
              "end": {
                "offset": 457,
                "line": 15,
                "column": 38
              }
            },
            "value": "\\"abc\\""
          }
        },
        {
          "kind": "Property",
          "loc": {
            "start": {
              "offset": 489,
              "line": 18,
              "column": 5
            },
            "end": {
              "offset": 522,
              "line": 18,
              "column": 37
            }
          },
          "readonly": true,
          "type": "var",
          "identifier": "someBool",
          "value": {
            "kind": "JavascriptBlock",
            "loc": {
              "start": {
                "offset": 523,
                "line": 18,
                "column": 39
              },
              "end": {
                "offset": 527,
                "line": 18,
                "column": 39
              }
            },
            "value": "true"
          }
        },
        {
          "kind": "Property",
          "loc": {
            "start": {
              "offset": 533,
              "line": 20,
              "column": 5
            },
            "end": {
              "offset": 572,
              "line": 20,
              "column": 32
            }
          },
          "typeModifier": "list",
          "type": "Rectangle",
          "identifier": "siblingRects"
        },
        {
          "kind": "Property",
          "loc": {
            "start": {
              "offset": 578,
              "line": 22,
              "column": 5
            },
            "end": {
              "offset": 614,
              "line": 22,
              "column": 40
            }
          },
          "typeModifier": "list",
          "type": "Rectangle",
          "identifier": "childRects",
          "value": {
            "kind": "ArrayBinding",
            "identifier": "childRects",
            "children": [
              {
                "kind": "ObjectDefinition",
                "identifier": "Rectangle",
                "children": [
                  {
                    "kind": "Attribute",
                    "identifier": "color",
                    "value": {
                      "kind": "JavascriptBlock",
                      "loc": {
                        "start": {
                          "offset": 644,
                          "line": 23,
                          "column": 28
                        },
                        "end": {
                          "offset": 649,
                          "line": 23,
                          "column": 28
                        }
                      },
                      "value": "\\"red\\""
                    }
                  }
                ]
              },
              {
                "kind": "ObjectDefinition",
                "identifier": "Rectangle",
                "children": [
                  {
                    "kind": "Attribute",
                    "identifier": "color",
                    "value": {
                      "kind": "JavascriptBlock",
                      "loc": {
                        "start": {
                          "offset": 680,
                          "line": 24,
                          "column": 28
                        },
                        "end": {
                          "offset": 686,
                          "line": 24,
                          "column": 28
                        }
                      },
                      "value": "\\"blue\\""
                    }
                  }
                ]
              }
            ]
          }
        },
        {
          "kind": "Function",
          "identifier": "foobar",
          "loc": {
            "start": {
              "offset": 699,
              "line": 27,
              "column": 5
            },
            "end": {
              "offset": 755,
              "line": 29,
              "column": 5
            }
          },
          "body": "function foobar() {\\n        console.log('foobar');\\n    }"
        }
      ]
    }
  ]
}~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

`;

exports[`import.qml - qml-verify: import.qml 1`] = `
import   QtQuick 2.5
import QtQuick.Window   2.2 as   Window   
import   "path/to/library.js  " as Library
import "   path/to/components"

Rectangle {
    property var helloWorld

    property var veryLongValueProperty: ApplaunchService.Service.getInstance().maintenance['enable']

       property var someNumber: {
           1.5
       }
    property  var someList: [1, 2, 'three', "four"]
    property var someObject: Rectangle { width: 100; height: 100; color: 'red' }
    property color   someColor:   "blue"

    // Default property
    default property var someString: "abc"

    // Read-only property
    readonly property var   someBool: true

    property   list<Rectangle> siblingRects

    property list<Rectangle> childRects: [
        Rectangle { color: "red" },
        Rectangle { color: "blue"}
    ]

    function foobar() {
        console.log ('foobar');
    }

}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
import QtQuick 2.5
import QtQuick.Window 2.2 as Window
import "path/to/library.js" as Library
import "path/to/components"
Rectangle {
    property var helloWorld
    property var veryLongValueProperty: {
        return ApplaunchService.Service.getInstance().maintenance["enable"];
    }
    property var someNumber: {
        1.5;
    }
    property var someList: [1, 2, "three", "four"];
    property var someObject: Rectangle {
        width: 100;
        height: 100;
        color: "red";

    }
    property color someColor: "blue";

    default property var someString: "abc";

    readonly property var someBool: true;
    property Rectangle siblingRects
    property Rectangle childRects:
    function foobar() {
        console.log("foobar");
    }
}
`;
